### ISSUE TITLE
Custom useFetch losing original event during SSR

### ISSUE THREAD

**Issue by shaunnbarron** [nuxt/nuxt#27750]
### Environment

- Operating System: `Darwin`
- Node Version:     `v20.15.0`
- Nuxt Version:     `3.12.2`
- CLI Version:      `3.12.0`
- Nitro Version:    `-`
- Package Manager:  `pnpm@9.4.0`
- Builder:          `-`
- User Config:      `devtools`
- Runtime Modules:  `-`
- Build Modules:    `-`

### Reproduction

https://stackblitz.com/edit/github-eycskg?file=app.vue

### Describe the bug

Creating a custom useFetch using the following docs:
https://nuxt.com/docs/guide/recipes/custom-usefetch#custom-usefetch

During SSR, the original `event` seems to be lost. Or the headers at least.

**Comment by OrbisK** [comment]
@cernymatej cant this be fixed with the new factory?

**Comment by cernymatej** [comment]
@OrbisK yeah good find! I'm pretty sure the old example didn't even use `useRequestFetch`. Anyway, the new custom instances behave exactly like regular `useFetch` so we can probably close this along with the draft PR.


Resolved by: https://github.com/nuxt/nuxt/pull/32300
